tests/test_checker.py

```diff
@@ -60,6 +60,82 @@ def test_spool_failure(self):
+upbackend,BACKEND,0,0,0,0,400,0,0,0,0,0,,0,0,0,0,UP,0,0,0,,1,259589,259589,,1,14,0,,0,,1,0,,0,,,,0,0,0,0,0,0,,,,,0,0,0,0,0,0,-1,,,0,0,0,0,
+postgres5503,FRONTEND,,,0,0,2000,0,0,0,0,0,0,,,,,OPEN,,,,,,,,,1,21,0,,,,0,0,0,0,,,,,,,,,,,0,0,0,,,0,0,0,0,,,,,,,,
+postgres-soa-slave,localhost,0,0,0,0,,0,0,0,,0,,0,0,0,0,DOWN,10,1,0,2,2,259401,259435,,1,22,1,,0,,2,0,,0,L4CON,,0,,,,,,,0,,,,0,0,,,,,-1,Connection refused,,0,0,0,0,
+postgres-soa-slave,BACKEND,0,0,0,0,200,0,0,0,0,0,,0,0,0,0,DOWN,0,0,0,,2,259401,259435,,1,22,0,,0,,1,0,,0,,,,,,,,,,,,,,0,0,0,0,0,0,-1,,,0,0,0,0,\n''')
+
+
+class InvalidHaproxyResponse(tornado.web.RequestHandler):
+    def get(self):
+        self.set_status(500)
+
+
+class ExceptionResponse(tornado.web.RequestHandler):
+    def get(self):
+        self.set_status(200)
+        self.write('\x00\x00')
+
+
+class TestHaproxyCheckerValidResponse(tornado.testing.AsyncHTTPTestCase):
+
+    def get_app(self):
+        return tornado.web.Application([
+            ('/;csv', ValidHaproxyResponse),
+        ])
+
+    @tornado.testing.gen_test
+    def test_downbackend(self):
+        response = yield checker.check_haproxy("downbackend", self.get_http_port(), "/", io_loop=self.io_loop, query_params="", headers={})
+        self.assertEqual((500, b'downbackend is DOWN'), response)
+
+    @tornado.testing.gen_test
+    def test_nonexistentbackend(self):
+        response = yield checker.check_haproxy("nonexistentbackend", self.get_http_port(), "/", io_loop=self.io_loop, query_params="", headers={})
+        self.assertEqual((500, b'nonexistentbackend is not found'), response)
+
+    @tornado.testing.gen_test
+    def test_upbackend(self):
+        response = yield checker.check_haproxy("upbackend", self.get_http_port(), "/", io_loop=self.io_loop, query_params="", headers={})
+        self.assertEqual((200, b'upbackend is UP'), response)
+
+
+class TestHaproxyCheckerInvalidResponse(tornado.testing.AsyncHTTPTestCase):
+
+    def get_app(self):
+        return tornado.web.Application([
+            ('/;csv', InvalidHaproxyResponse),
+        ])
+
+    @tornado.testing.gen_test
+    def test_invalidhttpresponse(self):
+        response = yield checker.check_haproxy("", self.get_http_port(), "/", io_loop=self.io_loop, query_params="", headers={})
+        self.assertEqual((500, b''), response)
+
+
+class TestHaproxyCheckerExceptionResponse(tornado.testing.AsyncHTTPTestCase):
+
+    def get_app(self):
+        return tornado.web.Application([
+            ('/;csv', ExceptionResponse),
+        ])
+
+    @tornado.testing.gen_test
+    def test_badresponse(self):
+        port = self.get_http_port()
+        response = yield checker.check_haproxy("", self.get_http_port(), "/", io_loop=self.io_loop, query_params="", headers={})
+        self.assertEqual((599, b'Unhandled exception line contains NULL byte  %d' % port), response)
```

Comment:
this needs to be "string-to-format".format(stuff).encode() to convert the string to bytes